src/main/java/io/supertokens/signingkeys/SigningKeys.java

```diff
@@ -126,7 +128,18 @@ public JWTSigningKeyInfo getStaticKeyForAlgorithm(JWTSigningKey.SupportedAlgorit
     }
 
     public KeyInfo getLatestIssuedDynamicKey() throws StorageQueryException, StorageTransactionLogicException {
-        return getDynamicKeys().get(0);
+        CoreConfig config = Config.getConfig(main);
+        List<KeyInfo> dynamicKeys = getDynamicKeys();
+
+        KeyInfo latest = dynamicKeys.get(0);
```

Comment:
are you sure that dynamicKeys is ordered based on latest first? Do we have a test for that?

Reply:
They are ordered, and there is a test (I've just updated it to be more specific, `getDynamicKeysReturnsOrdered`)